controllers/openstackconfiggenerator.go

```diff
@@ -630,3 +651,335 @@ func (r *OpenStackConfigGeneratorReconciler) verifyNodeResourceStatus(instance *
+		Name:         "fencing-config",
+		Namespace:    instance.Namespace,
+		Type:         common.TemplateTypeNone,
+		InstanceType: instance.Kind,
+		AdditionalTemplate: map[string]string{
+			"fencing.yaml": "/openstackconfiggenerator/config/common/fencing.yaml",
+		},
+		Labels:        cmLabels,
+		ConfigOptions: templateParameters,
+	}
+
+	common.GetTemplateData(fencingTemplate)
+
+	return common.GetTemplateData(fencingTemplate), nil
+}
+
+//
+// generate TripleoDeploy configmap with environment file containing predictible IPs
+//
+func (r *OpenStackConfigGeneratorReconciler) createTripleoDeployCM(
+	instance *ospdirectorv1beta1.OpenStackConfigGenerator,
+	cond *ospdirectorv1beta1.Condition,
+	envVars map[string]common.EnvSetter,
+	cmLabels map[string]string,
+	ospVersion ospdirectorv1beta1.OSPVersion,
+	controlPlane *ospdirectorv1beta1.OpenStackControlPlane,
+	tripleoTarballCM *corev1.ConfigMap,
+) (*corev1.ConfigMap, error) {
+	//
+	// generate OOO environment file with predictible IPs
+	//
+	templateParameters, rolesMap, err := openstackconfiggenerator.CreateConfigMapParams(r, instance, cond)
+	if err != nil {
+		cond.Message = err.Error()
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.ConfigGeneratorCondReasonRenderEnvFilesError)
+		cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.ConfigGeneratorError)
+		err = common.WrapErrorForObject(cond.Message, instance, err)
+
+		return nil, err
+	}
+
+	//
+	// get clusterServiceIP for fencing routing in the nic templates
+	//
+	clusterServiceIP, err := r.getClusterServiceEndpoint(
+		instance,
+		cond,
+		"default",
+		map[string]string{
+			"component": "apiserver",
+			"provider":  "kubernetes",
+		},
+	)
+	if err != nil {
+		cond.Message = err.Error()
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.ConfigGeneratorCondReasonClusterServiceIPError)
+		cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.ConfigGeneratorError)
+		err = common.WrapErrorForObject(cond.Message, instance, err)
+
+		return nil, err
+	}
+
+	//
+	// Render VM role nic templates, but only for tripleo roles
+	//
+	roleNicTemplates := r.createVMRoleNicTemplates(
+		instance,
+		ospVersion,
+		rolesMap,
+		clusterServiceIP,
+		cmLabels,
+	)
+
+	//
+	// Render fencing template
+	//
+	fencingTemplate, err := r.createFencingEnvironmentFiles(
+		instance,
+		cond,
+		controlPlane,
+		tripleoTarballCM,
+		cmLabels,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	common.MergeStringMaps(roleNicTemplates, fencingTemplate)
+
+	//
+	// create tripleo-deploy-config configmap with rendered data
+	//
+	cm := []common.Template{
+		{
+			Name:               "tripleo-deploy-config",
+			Namespace:          instance.Namespace,
+			Type:               common.TemplateTypeConfig,
+			InstanceType:       instance.Kind,
+			AdditionalTemplate: map[string]string{},
+			CustomData: common.MergeStringMaps(
+				roleNicTemplates,
+				fencingTemplate,
+			),
```

Comment:
Just curious as to why `common.MergeStringMaps` is called on 818 and 830 with the same parameters.

Reply:
Line 818 is nonsense, can not remember why it is there. probably I wanted to use a var, but for whatever reason stopped that. We can just remove 818.

Reply:
removed in new commit